### DISCUSSION TITLE
Support prompting on any commands that now fail due to no project name being provided

### DISCUSSION THREAD

**Discussion by sheldonhull** [miniscruff/changie#526] [Ideas] [answered]
I used to be able to script the changie workflow to minimize effort. For example:

```shell
#!/usr/bin/env bash
gum format "🚀 _changie bump and release_"
changie batch auto &&   # 👈👈👈👈👈👈👈 won't work now in mix of project + nonproject repos
  changie merge &&      # 👈👈👈👈👈👈👈 won't work now in mix of project + nonproject repos
  trunk fmt &&
  git add '.changes/*' &&
  git add CHANGELOG.md

gum confirm "Create Commit?" && git commit -m"$(gum write --value="feat: 🚀 release $(changie latest)")"
gum confirm "Tag?" && git tag -a $(changie latest) -F .changes/$(changie latest).md && gum format "🚀 _changie tagged to $(changie latest)_"
gum confirm "Sync Changes?" && git town sync

```

So here's what I'm running into now. 
The changie batch auto doesn't give me a dialogue to pick the project.

👉 When project is required then provide a prompt.

I thought about asking for a `changie projects` function that could list the projects to iterate through in my commands. In the meantime, I think that just using `yq` to read the yaml makes sense.

**Comment by miniscruff** [comment, marked as answer]
Yea, I wasn't quite sure if this was going to cause problems or not. Batch and merge currently do not have any prompts. I did not add any prompts to them as they have always been fully automated actions.

Your workflow definitely works great but yea won't work well with projects enabled. It looks like it could run great as a GitHub action as well but that won't work with prompts either way.

Note that projects will also have a different release notes path so your script will fail later as well.

I think having a gum select to first ask the project would be what I would do, then pass that into batch, merge and the release notes path.

I think a `changie config ...` command might be useful to search for stuff like project names, kind names, etc. Not sure though.

**Reply by miniscruff** [reply]
@sheldonhull did you find a way to make this any easier or have you given this any more thought?

**Reply by sheldonhull** [reply]
No. I’m just going to have to provide and handle with my tasks (mage, etc). 
not sure I have anything actionable right now other than the prompting options I mentioned. I’m fine with it as is for now. Feel free to close discussion.

**Reply by TheSpyder** [reply]
A `changie config` command to return the list of projects would be awesome - my team is considering wildcards and filtering to get that list at the moment. Turns out we don't have a changelog for every project in our monorepo 😂 

`PROJECTS := $(filter-out .changes/unreleased .changes/header.tpl.md,$(wildcard .changes/*))`

**Reply by miniscruff** [reply]
> A `changie config` command to return the list of projects would be awesome - my team is considering wildcards and filtering to get that list at the moment. Turns out we don't have a changelog for every project in our monorepo 😂 
> 
> `PROJECTS := $(filter-out .changes/unreleased .changes/header.tpl.md,$(wildcard .changes/*))`

My idea was to provide a basic output of the changie configuration in yaml or json, as well as maybe a template option for a go template. Then you can filter anything in the config. 

So something like
```bash
changie config --template '{{range .Projects}}{{.Key}}{{end}}'
```
Or you could take the json output and pass it to jq or whatever right.

**Reply by TheSpyder** [reply]
Yeah that looks great!

**Reply by miniscruff** [reply]
Actually I just remembered, @TheSpyder I believe you can just use yq https://github.com/mikefarah/yq
Something like:
```bash
yq '.projects[].key' .changie.yaml
```
should be just as effective as a config option + go template.

I will write up a documentation ticket to get something like this written down.

**Reply by TheSpyder** [reply]
Ah that's cool!
